biz.aQute.bndlib/src/aQute/bnd/build/Run.java

```diff
@@ -1,18 +1,59 @@
 package aQute.bnd.build;
 
+import static java.lang.invoke.MethodType.methodType;
+
 import java.io.File;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.Map;
 
 import aQute.bnd.osgi.Processor;
+import aQute.lib.exceptions.Exceptions;
 
 public class Run extends Project {
 
+	/**
+	 * So far we have not included the resolver code inside bndlib. This means
+	 * that anybody inside bndlib cannot rely on the resolver code present.
+	 * However, we get more functions that take advantage of this tool. The
+	 * following code looks if the resolving counterpart of Run is present in
+	 * the classloader. Since this is a 1:1 replacement for this class, we try
+	 * to load the resolving Bndrun.
+	 */
+	private final static MethodHandle createBndrun;
+
+	static {
+		MethodHandles.Lookup lookup = MethodHandles.publicLookup();
+		MethodType type = methodType(Run.class, Workspace.class, File.class);
+		MethodHandle handle = null;
+		try {
+			Class<?> clazz = Run.class.getClassLoader()
+				.loadClass("biz.aQute.resolve.Bndrun");
+			handle = lookup.findStatic(clazz, "createBndrun", type);
+		} catch (Throwable e) {
+			handle = null;
+		}
+		createBndrun = handle;
+	}
+
 	/**
 	 * Create a Run that will be stand alone if it contains -standalone. In that
 	 * case the given workspace is ignored. Otherwise, the workspace must be a
 	 * valid workspace.
 	 */
 	public static Run createRun(Workspace workspace, File file) throws Exception {
+		if (createBndrun == null)
```

Comment:
This does sort of create a behavior change in Run depending upon whether the Bndrun class is visible to the class loader. This will be true in the gradle plugin, the mave plugins and the bnd command. I am not sure how this will be true in Eclipse which has OSGi class loading behavior and the bndlib bundle does not import from the resolve bundle, so this code wont be able to class load the Bndrun class from the resolve bundle. So this seems a parity issue.

If we really want bndlib to have access to the resolve code, perhaps we should merge them into a single jar?

Reply:
Yes, maybe it is time. The resolver is such a cool part and all drivers rely on it. It starts to feel strained to exclude it. I've also though of add an RunFactory plugin but that also requires changes in many places.

Reply:
I am still not sure how this can work in Eclipse since bndlib in Eclipse will never be able to load the Bndrun class.

Reply:
Good to remind me, we can use the 'bnd-plugins' attribute on the exported package. See Auxilary in Central

Reply:
Never noticed Auxillary before.

Reply:
Been there for for more than 4 years ...